rclcpp/src/rclcpp/context.cpp

```diff
@@ -243,6 +244,24 @@ Context::init(
 
     weak_contexts_ = get_weak_contexts();
     weak_contexts_->add_context(this->shared_from_this());
+
+
+    std::lock_guard<std::recursive_mutex> lock (on_shutdown_callbacks_mutex_);
+
+    graph_listener_ = std::make_shared<graph_listener::GraphListener>(shared_from_this());
+
+    if (!graph_listener_->is_started()) {
+  // Register an on_shutdown hook to shutdown the graph listener.
+  // This is important to ensure that the wait set is finalized before
+  // destruction of static objects occurs.
```

Comment:
indent is off by 2 spaces?

```suggestion
    // Register an on_shutdown hook to shutdown the graph listener.
    // This is important to ensure that the wait set is finalized before
    // destruction of static objects occurs.
```